warehouse/integrations/snowflake/snowflake.go

```diff
@@ -521,63 +509,133 @@ func (sf *Snowflake) joinColumnsWithFormatting(columns []string, format string)
 	}, ",")
 }
 
-func (sf *Snowflake) sampleDuplicateMessages(
-	ctx context.Context,
-	db *sqlmw.DB,
-	mainTableName,
-	stagingTableName string,
-) ([]duplicateMessage, error) {
+func (sf *Snowflake) trackDuplicateMessageInfo(
+	ctx context.Context, db *sqlmw.DB, mainTableName,
+	stagingTableName, sortedColumns string,
+) error {
 	if !lo.Contains(sf.config.debugDuplicateWorkspaceIDs, sf.Warehouse.WorkspaceID) {
```

Comment:
[question] Is `workspaceID` too broad? Perhaps `destinationID` would be better?

Reply:
Initially, it was added only for `mattermost` use case for their workspace. We can probably extend it as well to destinationID.